packages/mermaid/src/diagrams/xychart/xychartRenderer.ts

```diff
@@ -0,0 +1,140 @@
+import { select } from 'd3';
+import { Diagram } from '../../Diagram.js';
+import * as configApi from '../../config.js';
+import { log } from '../../logger.js';
+import { configureSvgSize } from '../../setupGraphViewbox.js';
+import {
+  DrawableElem,
+  TextElem,
+  TextHorizontalPos,
+  TextVerticalPos,
+} from './chartBuilder/Interfaces.js';
+
+export const draw = (txt: string, id: string, _version: string, diagObj: Diagram) => {
+  function getDominantBaseLine(horizontalPos: TextHorizontalPos) {
+    return horizontalPos === 'top' ? 'hanging' : 'middle';
+  }
+
+  function getTextAnchor(verticalPos: TextVerticalPos) {
+    return verticalPos === 'left' ? 'start' : verticalPos === 'right' ? 'end' : 'middle';
+  }
+
+  function getTextTransformation(data: TextElem) {
+    return `translate(${data.x}, ${data.y}) rotate(${data.rotation || 0})`;
+  }
+  const conf = configApi.getConfig();
+
+  log.debug('Rendering xychart chart\n' + txt);
+
+  const securityLevel = conf.securityLevel;
+  // Handle root and Document for when rendering in sandbox mode
+  let sandboxElement;
+  if (securityLevel === 'sandbox') {
+    sandboxElement = select('#i' + id);
+  }
+  const root = sandboxElement ? sandboxElement : select('body');
+
+  const svg = root.select(`[id="${id}"]`);
+
+  const group = svg.append('g').attr('class', 'main');
+
+  const width = 700;
+  const height = 500;
+
+  // @ts-ignore: TODO Fix ts errors
+  configureSvgSize(svg, height, width, true);
+
+  svg.attr('viewBox', '0 0 ' + width + ' ' + height);
+
+  // @ts-ignore: TODO Fix ts errors
+  diagObj.db.setHeight(height);
+  // @ts-ignore: TODO Fix ts errors
+  diagObj.db.setWidth(width);
```

Comment:
You can set the db type, and use it without ts-ignores.

```ts
const db = diagObj.db as XYChartDB;
```

Reply:
Done

Reply:
Fixed